src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectManagementService.cs

```diff
@@ -0,0 +1,126 @@
+using Microsoft.SqlTools.ServiceLayer.Connection;
+using Microsoft.SqlTools.ServiceLayer.ObjectManagement.Requests;
+using Microsoft.SqlTools.Utility;
+
+namespace Microsoft.SqlTools.ServiceLayer.ObjectManagement
+{
+    /// <summary>
+    /// Main class for ObjectManagement Service functionality
+    /// </summary>
+    public class ObjectManagementService
+    {
+        private static Lazy<ObjectManagementService> objectManagementServiceInstance = new Lazy<ObjectManagementService>(() => new ObjectManagementService());
+        public static ObjectManagementService Instance => objectManagementServiceInstance.Value;
+
+        public static ConnectionService connectionService;
+
+        private IProtocolEndpoint serviceHost;
+
+        public ObjectManagementService() { }
+
+        /// <summary>
+        /// Internal for testing purposes only
+        /// </summary>
+        internal static ConnectionService ConnectionServiceInstance
+        {
+            get
+            {
+                if (connectionService == null)
+                {
+                    connectionService = ConnectionService.Instance;
+                }
+                return connectionService;
+            }
+            set
+            {
+                connectionService = value;
+            }
+        }
+
+        public void InitializeService(IProtocolEndpoint serviceHost)
+        {
+            this.serviceHost = serviceHost;
+            this.serviceHost.SetRequestHandler(RenameRequest.Type, HandleProcessRenameEditRequest);
+
+        }
+        internal Task HandleRequest<T>(RequestContext<T> requestContext, Func<Task> action)
+        {
+            return Task.Run(async () =>
+            {
+                try
+                {
+                    await action();
+                }
+                catch (Exception e)
+                {
+                    await requestContext.SendError(e);
+                }
+            });
+        }
+        /// <summary>
+        /// Method to handle the renaming operation
+        /// </summary>
+        /// <param name="requestParams">parameters which are needed to execute renaming operation</param>
+        /// <param name="requestContext">Request Context</param>
+        /// <returns></returns>
+        internal Task HandleProcessRenameEditRequest(RenameRequestParams requestParams, RequestContext<bool> requestContext)
+        {
+            return this.HandleRequest<bool>(requestContext, async () =>
+           {
+               Logger.Verbose("Handle Request in HandleProcessRenameEditRequest()");
+               ConnectionInfo connInfo;
+               try
+               {
+                   if (connectionService.TryFindConnection(
+                           requestParams.OwnerUri,
+                           out connInfo))
+                   {
+                       using (SqlConnection sqlConn = ConnectionService.OpenSqlConnection(connInfo, "RenamingDatabaseObjects"))
```

Comment:
I am actually not sure whether we should have different application names within sql tools service. since it is already the case, let's keep it consistent, please define a string constant in this class ` public const string ObjectManagementServiceApplicationName = "azdata-object-management";` and  use it here.

Reply:
So that it can be re-used for other places in this service later on if needed. You could probably make it private for now though since we don't actually have a use case for that yet.